sdk/state/state_test.go

```diff
@@ -17,29 +18,34 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func Test(t *testing.T) {
-	const horizonURL = "http://localhost:8000"
-	client := &horizonclient.Client{HorizonURL: horizonURL}
-	networkDetails, err := client.Root()
-	require.NoError(t, err)
-	networkPassphrase := networkDetails.NetworkPassphrase
+const horizonURL = "http://localhost:8000"
 
-	// Channel constants.
-	const observationPeriodTime = 20 * time.Second
-	const averageLedgerDuration = 5 * time.Second
-	const observationPeriodLedgerGap = int64(observationPeriodTime / averageLedgerDuration)
+var networkPassphrase string
+var client *horizonclient.Client
 
-	// Common data both participants will have during the test.
-	type Participant struct {
-		Name                 string
-		KP                   *keypair.Full
-		Escrow               *keypair.Full
-		EscrowSequenceNumber int64
-		Contribution         int64
+type Participant struct {
+	Name                 string
+	KP                   *keypair.Full
+	Escrow               *keypair.Full
+	EscrowSequenceNumber int64
+	Contribution         int64
+}
+
+// Setup
+func TestMain(m *testing.M) {
+	client = &horizonclient.Client{HorizonURL: horizonURL}
+	networkDetails, err := client.Root()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: %v\n", err)
+		os.Exit(1)
 	}
+	networkPassphrase = networkDetails.NetworkPassphrase
 
-	// Setup initiator.
-	initiator := Participant{
+	os.Exit(m.Run())
+}
+
+func initAccounts(t *testing.T) (initiator Participant, responder Participant) {
```

Comment:
style: I'm generally discouraging of naming the return types in the signature (would rather use a comment), because that requires you to do a `=` assignment instead of a `:=` assignment which requires a code reader to check if that variable has been assigned anywhere above and whether it is used etc, essentially taking away from the "don't reuse variables" philosophy.

(another code style pattern that we don't have to adopt but thought I'd bring up in case we find it useful and choose to adopt)

Reply:
I see your point thx for pointing out 🤔 . I'd be happy to discuss but the comment solution feels weird to me. I might be just not used to it though. I think when Leigh gets back who has more golang experience than I he can give his input here.

I think I'll leave as is for now if alright 😬